### DISCUSSION TITLE
Alexa Media Reauthentication Required

### DISCUSSION THREAD

**Discussion by cbhiii** [alandtse/alexa_media_player#1692] [Q&A]
Is anyone having issues with this integration reporting again and again that "**Alexa Media Reauthentication Required**". Over the last week or so it seems I'm having to do this once, twice or more times a day to get it to stay connected.

My assumption is this is an Amazon thing, but perhaps something _has changed_ with either HA or AMP recently?

Anyone else noticing this? Frustrating.

![Untitled](https://user-images.githubusercontent.com/10853381/185225211-406313bb-a133-4dc8-8fc9-c0a6196f50a3.jpg)

**Comment by BettySwallocks** [comment]
Same issue here, came up today - no guidance available that I can find on reauth process.

**Comment by mcrespov** [comment]
The same thing has been happening to me for two days, several times a day it is necessary to re-authenticate

**Comment by PontifexPrimus** [comment]
Same here, started yesterday and continued throughout the day today.

**Comment by kstockl** [comment]
The same here since today.

**Comment by H0d0rKF** [comment]
Same here since today. One action over Alexa Media Player and afterwards i have to reauthanticate.

**Comment by Zippoz** [comment]
Me to

**Comment by bonsaijogi** [comment]
I have the same problem, with me it started already 3 days ago

**Comment by tfranger** [comment]
I have the same problem, with me it started already 3 days ago

**Comment by ThijmenDam** [comment]
Same issue here. Just started looking into it, so haven't figured out the culprit. Will keep you posted if I find anything.

EDIT: Tried to re-authenticate by removing the integration and configuring it from scratch (i.e. regenerate 2FA credentials etc.), but so far no luck. In the logs I see the following:

` alexaapi.get_state((<alexapy.alexaapi.AlexaAPI object at 0x7f086ed60d60>,), {}): A login error occured: An exception of type JSONDecodeError occurred. Arguments: ('Expecting value: line 1 column 1 (char 0)',) `

Might it have something to do with the amazon domain that we have to provide during setup? I'm not exactly sure why this is necessary, and it would be great if anybody could elaborate.

See: https://github.com/custom-components/alexa_media_player/issues/1701

**Comment by KEV-github** [comment]
Same here for the past 2 days. HAOS on bare metal NUC mini,

**Comment by mauriziomontanaro** [comment]
Me too.. same problem!

**Comment by EmmanuelThe74** [comment]
@alandtse thank you for the last update, it seems to fix the issue of reauthentication, great job 

Guys, update your integration, it's currently stable for more than 24h finally

**Comment by ibanmda** [comment]
I have same issue. Not resolved with last upgrade 4.1.2.

**Reply by CommittotheIndian** [reply]
I'm still having this issue. Version 4.1.2 has
Not resolved it for me.

**Comment by mauriziomontanaro** [comment]
Also for me!

**Comment by mauriziomontanaro** [comment]
I solved by setting up the `external_url` and `internal_url` in my HA configuration.yaml, rebooted, retried the integration and it worked.

https://www.home-assistant.io/docs/configuration/basic/
